core/src/replay_stage.rs

```diff
@@ -767,28 +770,89 @@ impl ReplayStage {
             lockouts_sender,
         );
 
-        if let Some(ref voting_keypair) = voting_keypair {
-            let node_keypair = cluster_info.read().unwrap().keypair.clone();
+        Self::push_vote(
+            cluster_info,
+            bank,
+            vote_account_pubkey,
+            authorized_voter_keypairs,
+            tower.last_vote_and_timestamp(),
+            tower_index,
+        );
+        Ok(())
+    }
 
-            // Send our last few votes along with the new one
-            let vote_ix = vote_instruction::vote(
-                &vote_account,
-                &voting_keypair.pubkey(),
-                tower.last_vote_and_timestamp(),
-            );
+    fn push_vote(
+        cluster_info: &Arc<RwLock<ClusterInfo>>,
+        bank: &Arc<Bank>,
+        vote_account_pubkey: &Pubkey,
+        authorized_voter_keypairs: &[Arc<Keypair>],
+        vote: Vote,
+        tower_index: usize,
+    ) {
+        if authorized_voter_keypairs.is_empty() {
+            return;
+        }
 
-            let mut vote_tx =
-                Transaction::new_with_payer(vec![vote_ix], Some(&node_keypair.pubkey()));
+        let vote_state =
+            if let Some((_, vote_account)) = bank.vote_accounts().get(vote_account_pubkey) {
```

Comment:
This does cause a second decode of the vote account for each vote.  The other decode happens in `tower.new_vote_from_bank()`.  I don't see why this would be a perf issue, just pointing it out

Reply:
I added a method in the `Bank` that allows you to get the authorized voter by calling `bank.epoch_authorized_voter(vote_account_pubkey)`.

Reply:
But that only applies to vote accounts that have stake in the epoch right?

Reply:
ah shoot yeah, you're right :P